impl/src/main/java/org/jboss/cdi/tck/tests/full/context/passivating/dependency/builtin/BuiltinBeanPassivationDependencyTest.java

```diff
@@ -97,7 +97,11 @@ public void testInjectionPoint() throws IOException, ClassNotFoundException {
         assertEquals(inspectorCopy.getInjectionPoint().getQualifiers(), inspector.getInjectionPoint().getQualifiers());
         assertEquals(inspectorCopy.getInjectionPoint().getBean(), inspector.getInjectionPoint().getBean());
         assertEquals(inspectorCopy.getInjectionPoint().getMember(), inspector.getInjectionPoint().getMember());
-        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated(), inspector.getInjectionPoint().getAnnotated());
+
+        // Annotated does not necessarly implement equals()/hashcode()
+        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated().getBaseType(), inspector.getInjectionPoint().getAnnotated().getBaseType());
+        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated().getAnnotations(), inspector.getInjectionPoint().getAnnotated().getAnnotations());
```

Comment:
These are both good, but I think we should also (mainly?) assert that the underlying reflection objects are equal (which should work?). I know we already assert equality on `InjectionPoint.getMember()`, but I do think we should have both.

Something like

```java
assertEquals(
    unwrap(inspectorCopy.getInjectionPoint().getAnnotated()),
    unwrap(inspector.getInjectionPoint().getAnnotated())
);
```

where

```java
private static Object unwrap(Annotated annotated) {
    if (annotated instanceof AnnotatedMember) {
        return ((AnnotatedMember) annotated).getJavaMember();
    } else if (annotated instanceof AnnotatedParameter) {
        return ((AnnotatedParameter) annotated).getJavaParameter();
    } else if (annotated instanceof AnnotatedType) {
        return ((AnnotatedType) annotated).getJavaClass();
    } else {
        throw new UnsupportedOperationException("Unknown Annotated instance: " + annotated);
    }
}
```

WDYT?

Reply:
Oh I see.
I'm happy to update the PR according to this. It looks more complete.